sunpy/data/data_manager/cache.py

```diff
@@ -107,7 +108,14 @@ def _has_expired(self, details):
         `bool`
             Whether the url has expired or not.
         """
-        return self._expiry and datetime.now() - datetime.fromisoformat(details['time']) > self._expiry
+        time = details.get("time", datetime.now().isoformat())
+
+        # TODO: Remove this once we depend on Python >=3.7
```

Comment:
What's this magic?

Reply:
What bit?

Reply:
Is fromisoformat not present in some python versions? I thought that was always there. Interesting.

Reply:
It was added in 3.7.